src/components/controls/language.js

```diff
@@ -16,8 +16,18 @@ import { CHANGE_LANGUAGE } from "../../actions/types";
   };
 })
 class Language extends React.Component {
+  async ensureLanguageResources(lang) {
+    for (const ns of ["language", "sidebar", "translation"]) {
+      if (!i18n.hasResourceBundle(lang, ns)) {
+        const res = await import(/* webpackMode: "lazy-once" */ `../../locales/${lang}/${ns}.json`); // eslint-disable-line
```

Comment:
So basically you have 3 scenarios:

1) Without an explicit locale config in webpack.config.js, and with no `webpackMode: "lazy-once"`, each locale will create its own micro-chunk. This may seem a good idea at first glance because most people are only ever going to use one locale, however what happens is that various chunk merging plugins tend to merge these micro-chunks in very unpredictable ways and you get the locales bundled up with other stuff, defeating the purpose. So I would advise against this.

2) With `lazy-once` and the explicit single-package locale config (as is now), the `lazy-once` is basically ignored, it only serves as a safeguard against scenario (1) above.

3) If you decide to split locales further, e.g.
```
          locales1: {
            test: /\/src\/locales\//,
            name: "locales-others",
            enforce: true,
            priority: 1,
            chunks: "all"
          },
          locales2: {
            test: /\/src\/locales\/(es|pt|fr)\//,
            name: "locales-es.pt.fr",
            enforce: true,
            priority: 2, // This priority must be higher than the "locales-others chunk
            chunks: "all"
          },
```,
in that case you should absolutely *remove* the `lazy-once` directive because otherwise when the locales get lazy loaded, all the separate packages will be loaded at once, thus nullifying the benefit of splitting them.